sw/device/tests/aes_force_prng_reseed_test.c

```diff
@@ -0,0 +1,509 @@
+}
+
+status_t execute_test(void) {
+  // Initialize AES, EDN, CSRNG, and Entropy Source
+  #if RESEED_TEST_EN
+    LOG_INFO("Test with RESEED_TEST_EN ");
+  #endif
+  #if DEBUG_EN
+    LOG_INFO("Test with DEBUG_EN ");
+  #endif
+  #if TRYSETTING1_EN
+    LOG_INFO("Test with TRYSETTING1_EN ");
+  #endif
+
+  // Initialize all modules
+  LOG_INFO("Initializing AES unit.");
+  CHECK_DIF_OK(dif_aes_init(mmio_region_from_addr(TOP_EARLGREY_AES_BASE_ADDR),
+                            &aes));
+  CHECK_DIF_OK(dif_aes_reset(&aes)); // Reset AES to a known state
+  LOG_INFO("Module initialized: AES");
+                                    
+  #if RESEED_TEST_EN
+    LOG_INFO("Starting AES PRNG reseed stall test with extra debug...");
+    CHECK_DIF_OK(dif_edn_init(mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR), &edn0));
+    CHECK_DIF_OK(dif_edn_init(mmio_region_from_addr(TOP_EARLGREY_EDN1_BASE_ADDR), &edn1));
+    CHECK_DIF_OK(dif_csrng_init(mmio_region_from_addr(TOP_EARLGREY_CSRNG_BASE_ADDR), &csrng));
+    CHECK_DIF_OK(dif_entropy_src_init(mmio_region_from_addr(TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR), &entropy_src));
+    LOG_INFO("Modules initialized: EDN0, EDN1, CSRNG, Entropy Source");
+  #endif
+
+  // Configure AES for ECB mode, 128-bit key, and manual operation
+  dif_aes_transaction_t transaction = {
+    .operation =
+        kDifAesOperationEncrypt,  // Set the AES operation to encryption
+    .mode =
+        kDifAesModeEcb,  // Use Electronic Codebook (ECB) mode for encryption
+    .key_len = kDifAesKey128,  // Use a 128-bit key for encryption
+    //.key_len = kDifAesKey256,  // Use a 256-bit key for encryption
+    .key_provider = kDifAesKeySoftwareProvided,  // The encryption key is
+                                                 // provided by software
+    //.mask_reseeding = kDifAesReseedPer64Block,   // Configure the PRNG to reseed
+    //                                             // every 64 blocks
+    .mask_reseeding = kDifAesReseedPerBlock,  // Resseed PRNG every block (128 bits)                                                 
+    //operation, where it starts processing as soon as data is available
+    //.manual_operation = kDifAesManualOperationAuto,  // Set AES to automatic
+    .manual_operation =
+        kDifAesManualOperationManual,  // Set AES to manual operation, where
+                                       // software manually triggers the
+                                       // operation
+   .reseed_on_key_change = true,  // Enable PRNG reseeding every time the key
+                                  // is changed (KEY_TOUCH_FORCES_RESEED)
+   // Setting force_masks to false as per disucsseion
+   .force_masks = false,     // Enable forced masking for the key (important for
+                            // security during testing)
+   .ctrl_aux_lock = false,  // Do not lock the auxiliary control settings,
+                             // allowing further modifications
+  };
```

Comment:
Assembly style comments are also not aligned with the style guide. Many of the comments here can be removed as they don't add any explanation over the enum names.